src/main/java/org/kaazing/netx/http/internal/HttpURLConnectionHandler.java

```diff
@@ -250,76 +248,85 @@ public InputStream getInputStream() throws IOException {
+                                values = new LinkedList<String>();
+                                cookies.put(name, values);
+                            }
+                            values.add(value);
                         }
-                        values.add(value);
-                    }
-                    else if ("WWW-Authenticate".equalsIgnoreCase(name)) {
-                        if (challenges == null) {
-                            challenges = new LinkedList<String>();
+                        else if ("WWW-Authenticate".equalsIgnoreCase(name)) {
+                            if (challenges == null) {
+                                challenges = new LinkedList<String>();
+                            }
+                            challenges.add(value);
+                        }
+                        else {
+                            connection.addHeaderField(name, value);
                         }
-                        challenges.add(value);
                     }
-                    else {
-                        connection.addHeaderField(name, value);
+
+                    if (cookies != null && !cookies.isEmpty()) {
+                        CookieHandler handler = CookieHandler.getDefault();
+                        if (handler != null) {
+                            connection.storeCookies(handler);
+                        }
                     }
-                }
 
-                if (cookies != null && !cookies.isEmpty()) {
-                    CookieHandler handler = CookieHandler.getDefault();
-                    if (handler != null) {
-                        connection.storeCookies(handler);
+                    state = State.HANDSHAKE_RECEIVED;
+
+                    switch (responseCode) {
+                    case HTTP_SWITCHING_PROTOCOLS:
+                    case HTTP_MOVED_PERM:
+                    case HTTP_MOVED_TEMP:
+                    case HTTP_SEE_OTHER:
+                        break;
+                    case HTTP_UNAUTHORIZED:
+                        // Note: check maximum attempts
+                        processChallenges(challenges);
+                        break;
+                    default:
+                        throw new IllegalStateException(format("Upgrade failed (%d)", responseCode));
                     }
                 }
-
-                state = State.HANDSHAKE_RECEIVED;
-
-                switch (responseCode) {
-                case HTTP_SWITCHING_PROTOCOLS:
-                case HTTP_MOVED_PERM:
-                case HTTP_MOVED_TEMP:
-                case HTTP_SEE_OTHER:
-                    break;
-                case HTTP_UNAUTHORIZED:
-                    // Note: check maximum attempts
-                    processChallenges(challenges);
-                    break;
-                default:
-                    throw new IllegalStateException(format("Upgrade failed (%d)", responseCode));
+                finally {
+                    reader.close();
```

Comment:
LineReader.close() is a no-op for precisely that reason. The only reason I had to do that was to make Eclipse happy. Otherwise, it shows warning like "Resource leak: reader not closed".